addons_tests/test_module/models/employee_contract.py

```diff
@@ -0,0 +1,41 @@
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
+
+
+class employee_contract(models.Model):
+    _inherit = 'hr.employee'
+
+    contract_ids = fields.One2many('hop.dong', 'id', string='contract_ids')
+
+    status = fields.Selection([
+        ('draft', 'Draft'),
+        ('waiting_approve', 'Waiting Approve'),
+        ('approved', 'Approved'),
+        ('terminated', 'Terminated')],
+        string='Status', default='draft')
+
+    work_experience_ids = fields.One2many('employee.work.exp', 'employee_id')
+
+    def action_draft(self):
+        self.status = 'draft'
+
+    def action_waiting_approve(self):
+        self.status = 'waiting_approve'
+
+    def action_approved(self):
+        self.status = 'approved'
+
+    def action_terminated(self):
+        self.status = 'terminated'
+
+    @api.multi
+    def action_open_employee_contract(self):
+        action = self.env.ref('test_module.action_hop_dong').read()[0]
+        action['domain'] = [('employee_id', '=', self.id)]
+        return action
+
+    @api.multi
+    def action_approve(self):
+        for record in self:
```

Comment:
chỗ này gọi lại hàm action_approved có được không nhỉ?

Reply:
gọi được nhưng sẽ là vòng lặp vô hạn ạ

Reply:
> c nhưng sẽ là vòng lặp vô hạn

sao lại vô hạn em nhỉ?

Reply:
> nếu gọi chính nó trong vòng for thì sẽ vô hạn a

hàm hiện tại là action_approve còn anh bảo gọi lại hàm action_approved. 2 hàm khác nhau sao lại gọi chính nó?

Reply:
ở đây gọi lại được action_approved, nhưng xóa vòng for ạ